src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -86,10 +94,56 @@ public async Task DeployCdkProject(OrchestratorSession session, string cdkProjec
                 redirectIO: true,
                 streamOutputToInteractiveService: true);
 
+            await CheckCdkDeploymentFailure(cloudApplication);
```

Comment:
Currently you are fetching **all** events associated with the CF stack We need to find the DateTime of the last event that occurred before the [cdkDeploy command](https://github.com/aws/aws-dotnet-deploy/pull/433/files#diff-1f36ff037bca7e527df601920aa6f58376c0e7aceb08c49238d40e9f8e2836beR90) is invoked and then pass it to the `CheckCdkDeploymentFailure` method. Inside this method, we only need to inspect events that occurred after the `lastEventTime`. If the CF stack is brand new, then set the `lastEventTime` to `DateTime.Now`

Reply:
Fixed